Src/PerformanceCollector/Shared/Implementation/IPerformanceCollector.cs

```diff
@@ -31,5 +30,12 @@ internal interface IPerformanceCollector
         /// <param name="error">Captures the error logged.</param>
         /// <param name="blockCounterWithInstancePlaceHolder">Boolean that controls the registry of the counter based on the availability of instance place holder.</param>
         void RegisterCounter(string perfCounter, string reportAs, bool isCustomCounter, out string error, bool blockCounterWithInstancePlaceHolder);
+
+        /// <summary>
+        /// Removes a counter.
+        /// </summary>
+        /// <param name="perfCounter">Name of the performance counter to remove.</param>
+        /// <param name="reportAs">ReportAs value of the counter to remove.</param>
+        void RemoveCounter(string perfCounter, string reportAs);
```

Comment:
Can every UX has it's own configuration then? You can create a new configuration and switch to it only after it was fully enabled. Instead of merging all streams and managing all different stream IDs you can just make each stream be responsible for collecting data needed for it.